parts/linux/cloud-init/artifacts/cse_main.sh

```diff
@@ -268,6 +268,15 @@ if [ "${NEEDS_CONTAINERD}" == "true" ]; then
 [Service]
 Environment="KUBELET_CONTAINERD_FLAGS=--container-runtime=remote --runtime-request-timeout=15m --container-runtime-endpoint=unix:///run/containerd/containerd.sock --runtime-cgroups=/system.slice/containerd.service"
 EOF
+    
+    # --container-runtime flag is removed in 1.27
+    # therefore, do not pass it to kubelet for k8s >= 1.27
+    if semverCompare ${KUBERNETES_VERSION:-"0.0.0"} "1.27.0"; then
+        tee "/etc/systemd/system/kubelet.service.d/10-containerd.conf" > /dev/null <<'EOF'
+[Service]
+Environment="KUBELET_CONTAINERD_FLAGS=--runtime-request-timeout=15m --container-runtime-endpoint=unix:///run/containerd/containerd.sock --runtime-cgroups=/system.slice/containerd.service"
```

Comment:
are runtime request and runtime cgroups actually deprecated? or only runtime-endpoint?

Reply:
both runtime request and runtime cgroups present in 1.27

![image](https://github.com/Azure/AgentBaker/assets/26747662/4ae76410-86df-4be5-98df-c09c70488ca4)

Reply:
the only deprecated flag is --container-runtime=remote